test/servinge2e/servicemesh_test.go

```diff
@@ -190,3 +231,295 @@ func TestKsvcWithServiceMeshSidecar(t *testing.T) {
+	if err != nil {
+		return "", fmt.Errorf("error signing JWT token with PKSC1v15: %v", err)
+	}
+
+	jwtSignatureBase64 := base64.RawURLEncoding.EncodeToString(signature)
+	jwtToken := jwtSigningInput + "." + jwtSignatureBase64
+
+	return jwtToken, nil
+}
+
+// Convenience method to test requests with tokens, reads the response and returns a closed response and the body bits
+func jwtHttpGetRequestBytes(url, token string) (*http.Response, []byte, error) {
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error creating HTTP GET request: %v", err)
+	}
+	req.Header.Add("Authorization", "Bearer "+token)
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error doing HTTP GET request: %v", err)
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	return resp, body, err
+}
+
+// Verifies access to a Ksvc with an istio-proxy can be configured
+// via istio authentication Policy to allow valid JWT only.
+// Skipped unless ServiceMesh has been installed via "make install-mesh"
+func TestKsvcWithServiceMeshJWTDefaultPolicy(t *testing.T) {
+
+	caCtx := test.SetupClusterAdmin(t)
+	test.CleanupOnInterrupt(t, func() { test.CleanupAll(t, caCtx) })
+	defer test.CleanupAll(t, caCtx)
+
+	// Skip test if ServiceMesh not installed
+	assumeServiceMeshInstalled(caCtx)
+
+	privateKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		t.Fatalf("error generating private key: %v", err)
+	}
+
+	// print out the public key for debugging purposes
+	publicPksvc1, err := x509.MarshalPKIXPublicKey(&privateKey.PublicKey)
+	publicPem := pem.EncodeToMemory(&pem.Block{
+		Type:  "RSA PUBLIC KEY",
+		Bytes: publicPksvc1,
+	})
+	t.Logf("%s", string(publicPem))
+
+	// Used as "kid" in JWKS
+	const keyId = "test"
+	const issuer = "testing-issuer@secure.serverless.openshift.io"
+	const subject = "testing-subject@secure.serverless.openshift.io"
+	// For testing an invalid token (with a different issuer)
+	const wrongIssuer = "eve@secure.serverless.openshift.io"
+
+	jwks, err := rsaPublicKeyAsJwks(privateKey.PublicKey, keyId)
+	if err != nil {
+		t.Fatalf("error encoding RSA public key as JWKS: %v", err)
+	}
+
+	// Istio 1.1 and earlier lack "jwks" option, only "jwksUri", so we need to host it on some URL
+	// We'll misuse the "hello-openshift" image with the JWKS file defined as the RESPONSE env, and deploy this as a ksvc
+	jwksKsvc := test.Service("jwks", serviceMeshTestNamespaceName, "openshift/hello-openshift", nil)
+	jwksKsvc.Spec.Template.Spec.Containers[0].Env = append(jwksKsvc.Spec.Template.Spec.Containers[0].Env, core.EnvVar{
+		Name:  "RESPONSE",
+		Value: jwks,
+	})
+	jwksKsvc.ObjectMeta.Labels = map[string]string{
+		serving.VisibilityLabelKey: serving.VisibilityClusterLocal,
+	}
+	jwksKsvc = withServiceReadyOrFail(caCtx, jwksKsvc)
+
+	// Create a Policy
+	authentication := &unstructured.Unstructured{
```

Comment:
probably, a better name would be `authPolicy` . ...authentication is too general and overloaded